### PR TITLE
box: role 'public' rw access on _session_settings

### LINKED ISSUE #6310: Newly created user doesn't have access to box.space._session_settings
Version: 2.9.0-253-gdd7fa342997d

Steps to reproduce:

```
tarantool> box.schema.user.create('test')
---
...

tarantool> box.session.su('test')
---
...

tarantool> box.space._session_settings:select()
---
- error: Read access to space '_session_settings' is denied for user 'test'
...
```

Interestingly, the user still can change SQL features:

```
tarantool> box.execute('set session "sql_vdbe_debug" = true;')
---
- row_count: 1
...
```

Since `box.space._session_settings` is a virtual space, which only contains information about the current session, it should be accessible by anyone without granting any additional privileges.

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
The _session_settings space is a virtual space which only contains
information about the current session, hence, it should be accessible
by everyone without granting any additional privileges. New users are
granted 'public' role by default: grant read, write access on space
_session_settings to role 'public'.

Closes #6310

### CODE REVIEW

File: src/box/lua/upgrade.lua [hunk 1]
```diff
@@ -1208,8 +1208,17 @@ local function remove_sql_builtin_functions_from_func()
...
+    local _session_settings = box.space[box.schema.SESSION_SETTINGS_ID]
+    local _priv = box.space[box.schema.PRIV_ID]
+    log.info("grant read,write access on _session_settings space to public role")
+    _priv:insert({ADMIN, PUBLIC, 'space', _session_settings.id,
+                  box.priv.R + box.priv.W})
+end
+
 local function upgrade_to_2_9_1()
     remove_sql_builtin_functions_from_func()
+    grant_rw_access_on__session_settings_to_role_public()
```
Comment (Korablev77) [inline src/box/lua/upgrade.lua:1221]: This should be part of 2.10 upgrade: 2.9 is current version, so users may have already called this function..

File: test/box-luatest/gh_6310_grant_rw_access_on__session_settings_space_to_public_role_test.lua [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    end)
+    t.assert(privs and privs:find('read,write'), msg_fmt:format('bootstrapped'))
+end
+
+g.test_upgrade = function()
+    local privs = g.upgrade:exec(function()
+        box.schema.upgrade()
+        local public_privs = box.schema.role.info('public')
+        for _, priv in pairs(public_privs) do
+            if priv[3] == '_session_settings' then
```
Comment (Korablev77) [inline test/box-luatest/gh_6310_grant_rw_access_on__session_settings_space_to_public_role_test.lua:51]: I'd also add original test from the issue to make sure that privileges work fine:)

File: src/box/lua/upgrade.lua (hunk 1 above)
Comment (CuriousGeorgiy) [reply src/box/lua/upgrade.lua:1221]: @Korablev77 actually, the current version of Tarantool's `HEAD` seems to be 2.9.0:

```bash
./tarantool -v
Tarantool 2.9.0-682-g310082478
Target: Linux-x86_64-Debug
```

Comment (Korablev77) [reply src/box/lua/upgrade.lua:1221]: On master branch:
```
./src/tarantool -v
Tarantool 2.10.0-beta1-317-gc9787d572
Target: Linux-x86_64-Debug
```

Comment (CuriousGeorgiy) [reply src/box/lua/upgrade.lua:1221]: @Korablev77 weird... I am pulling master, as you can see by the commit hash, perhaps, something is wrong with my fork

```bash
git describe --tags
2.9.0-689-gc9787d572
```

Comment (CuriousGeorgiy) [reply src/box/lua/upgrade.lua:1227]: Fixed: changed to upgrade to 2.10.0. Sorry for the inconvenience.

**Review by alyapunov — APPROVED** [review]

**Review by Korablev77 — APPROVED** [review]

**Pull request merged**